[Bugfix] 1.7.10 version 404 response

YoYoK3nny opened the pull request:
Due to the way Forge versions for MC 1.7.10 are schemed, trying to download the Forge installer for 1.7.10 fails with a 404 response.

The changes in this pull request should fix that.

Also adds the ability to choose a promo flavor, defaulting to the recommended version like the origiinal code tends to do.

Note: I changed the filename parsing so that the downloaded file always has a consistent versioning scheme, versus what happened with 1.7.10 (at least filename wise)

Built and tested working.

YoYoK3nny committed:
[Bugfix] 1.7.10 version 404 response

Due to the way Forge versions for MC 1.7.10 are schemed, trying to download the Forge installer for 1.7.10 fails with a 404 response.

The changes in this pull request should fix that.

Also adds the ability to choose a promo flavor, defaulting to the recommended version like the origiinal code tends to do.

Note: I changed the filename parsing so that the downloaded file always has a consistent versioning scheme, versus what happened with 1.7.10 (at least filename wise)

Built and tested working.

itzg commented on main.go:
```diff
@@ -48,9 +48,14 @@ func run(c *cli.Context) error {
 	}
 
 	version := c.Args()[0]
+	latest := false
```
Could just do this, right?
```suggestion
	latest := c.Bool("latest")
```

YoYoK3nny replied:
Oh yeah, that is a bit faster for processing it. Need to remove the related if block in that function so I'll edit it manually. (Didn't know this change suggestion stuff existed, neat feature)

itzg commented on main.go:
```diff
@@ -113,6 +121,9 @@ func locatePromoVersion(mcVersion string) (error, string) {
 // the path to the downloaded file
 func download(mcVersion, promoVersion string) (error, string) {
 	combinedVer := fmt.Sprintf("%s-%s", mcVersion, promoVersion)
+	if mcVersion == "1.7.10" {
+		combinedVer = fmt.Sprintf("%s-%s-%s", mcVersion, promoVersion, mcVersion)
+	}
```
Oh yeah, I forgot they used to do that weird naming. Was it just 1.7.10 or all versions before that too?

YoYoK3nny replied:
From what I've seen and experienced with it, only 1.7.10 got this weird naming.
Even the promotions file that this downloads has a bit of weirdness in it that made me remove 1 of the for loops when I added the "latest" flag. It kept using the wrong entry.

itzg reviewed (commented):
Thanks! Just a suggestion and question, but I'm good to merge this when you're ready.

YoYoK3nny committed:
Optimized code for getting if the latest flag was specified.

Set the variable directly with the flag getter.

Removed related if block as no longer needed with this change.

YoYoK3nny commented:
Okay, it's ready to go.

EDIT1: On a side note, that "latest" variable only gets sent to locatePromoVersion()

Would it be better to just do this:

`err, promoVersion := locatePromoVersion(version, c.Bool("latest"))`

EDIT2: That works, so I'm gonna go with that too.

YoYoK3nny committed:
Further code optimization

Removed variable that only gets used once in favor of directly passing the flag to the function it was added for.

itzg commented:
[1.1.0](https://github.com/itzg/forge-downloader/releases/tag/1.1.0) includes your change.
